cmd/apps/kuma_app.go

```diff
@@ -0,0 +1,117 @@
+package apps
+
+import (
+	"fmt"
+	"github.com/alexellis/arkade/pkg"
+	"github.com/alexellis/arkade/pkg/apps"
+	"github.com/alexellis/arkade/pkg/config"
+	"github.com/alexellis/arkade/pkg/k8s"
+	"github.com/alexellis/arkade/pkg/types"
+	"github.com/spf13/cobra"
+)
+
+func MakeInstallKuma() *cobra.Command {
+	var kuma = &cobra.Command{
+		Use:          "kuma",
+		Short:        "Install Kuma",
+		Long:         "Install Kuma",
+		Example:      ` arkade app install kuma`,
+		SilenceUsage: true,
+	}
+
+	kuma.Flags().String("namespace", "kuma-system", "Namespace for the app")
+
+	kuma.Flags().String("control-plane-mode", "standalone", "Kuma CP modes: one of standalone,zone,global")
+	kuma.Flags().Bool("auto-scale", false, "Enable Horizontal Pod Autoscaling (requires the Metrics Server)")
+
+	kuma.Flags().Bool("use-cni", false, "Use CNI instead of proxy init container")
+	kuma.Flags().Bool("ingress", false, "Deploy Ingress for cross cluster communication")
+
+	kuma.Flags().StringArray("set", []string{}, "Use custom flags or override existing flags \n(example --set ingress.enabled=false)")
+
+	kuma.RunE = func(command *cobra.Command, args []string) error {
+		kubeConfigPath, _ := command.Flags().GetString("kubeconfig")
+		if err := config.SetKubeconfig(kubeConfigPath); err != nil {
+			return err
+		}
+
+		arch := k8s.GetNodeArchitecture()
+		fmt.Printf("Node architecture: %q\n", arch)
+		if suffix := getValuesSuffix(arch); suffix == "-armhf" {
+			return fmt.Errorf(`Kuma is currently not supported on armhf architectures`)
+		}
+
+		namespace, _ := command.Flags().GetString("namespace")
```

Comment:
Please add a PreRunE in a follow-up PR to validate all booleans "get" correctly - simply by getting them and checking there is no error.

https://github.com/alexellis/arkade/blob/master/cmd/apps/falco.go#L32

This can be done in a follow-up PR.

Reply:
Sure, I think I'll go through all the apps and add it if missing. (I'll raise an issue ASAP.)

Reply:
Just for this app is fine, since that's the one you're working on now.

Reply:
`PreRunE` added, also moved some validation logic there, since it makes more sense there then in `RunE`. :+1: